dali/operators/coord/coord_transform.h

```diff
@@ -0,0 +1,307 @@
+      if (is_fused)
+        Repeat(per_sample_translation_, translation_, N);
+    } else if (HasMatrixInputArg()) {
+      const auto &M_inp = ws.ArgumentInput(name);
+      auto M = view<const float>(M_inp);
+      DALI_ENFORCE(is_uniform(M.shape), "Matrices for all samples int the batch must have "
+        "the same shape.");
+      DALI_ENFORCE(M.shape.sample_dim() == 0 || M.shape.sample_dim() == 2, make_string(
+        "The parameter M must be a list of 2D matrices of same size or a list of scalars. Got: ",
+        M.shape));
+      if (M.shape.sample_dim() == 0) {  // we have a list of scalars - put them on diagonals
+        output_pt_dim_ = cols;
+        int mat_size = output_pt_dim_ * cols;
+        per_sample_mtx_.resize(mat_size * N);
+        for (int i = 0; i < N; i++) {
+          FillDiag(make_span(&per_sample_mtx_[i * mat_size], mat_size), M.data[i][0]);
+        }
+        translation_.resize(output_pt_dim_, 0);
+        Repeat(per_sample_translation_, translation_, N);
+      } else {
+        DALI_ENFORCE(M.shape[0][1] == cols, make_string("The shape of the argument ``", name,
+           "`` does not match the input shape. Got ", M.shape[0], " matrices and the input "
+           "requires matrices with ", cols, " columns"));
+        output_pt_dim_ = M.shape[0][0];
+        int mat_size = output_pt_dim_ * input_pt_dim_;
+        per_sample_mtx_.resize(mat_size * N);
+        if (is_fused)
+          per_sample_translation_.resize(output_pt_dim_ * N);
+        for (int s = 0; s < N; s++) {
+          for (int i = 0; i < output_pt_dim_; i++) {
+            for (int j = 0; j < input_pt_dim_; j++)
+                per_sample_mtx_[s * mat_size + i * input_pt_dim_ + j] = M.data[s][i * cols + j];
+            if (is_fused)
+              per_sample_translation_[s * output_pt_dim_ + i] = M.data[s][i * cols + cols - 1];
+          }
+        }
+      }
+    } else {
+      output_pt_dim_ = input_pt_dim_;
+      if (static_cast<int>(mtx_.size()) != output_pt_dim_ * input_pt_dim_) {
+        mtx_.resize(output_pt_dim_ * input_pt_dim_, 0);
+        FillDiag(mtx_, 1);
+        Repeat(per_sample_mtx_, mtx_, N);
+        if (is_fused) {
+          translation_.resize(output_pt_dim_, 0);
+          Repeat(per_sample_translation_, translation_, N);
+        }
+      }
+    }
+  }
+
+  void SplitFusedMT() {
+    int cols = input_pt_dim_ + 1;
+    translation_.resize(output_pt_dim_);
+    for (int i = 0; i < output_pt_dim_; i++) {
+      // store the last column in the translation vector
+      translation_[i] = mtx_[i * cols + input_pt_dim_];
+
+      // compact the matrix
+      if (i > 0) {
+        for (int j = 0; j < input_pt_dim_; j++)
+          mtx_[i * input_pt_dim_ + j] = mtx_[i * cols + j];
+      }  // else: the first row is already where it should be
+    }
+    mtx_.resize(output_pt_dim_ * input_pt_dim_);
+  }
+
+  /** @brief Fill the diagonal with a scalar value, put zeros elsewhere */
```

Comment:
Maybe it should be in a utility header?

Reply:
I'll make a separate PR with matrix argument handling and move much more to the utilities. Is that OK?

Reply:
I think you can do it here.